select_statements.sql

```diff
@@ -20,10 +20,45 @@ WHERE vehicle.stationName = "Uppsala station" AND booking.endDatum < "2024-04-13
 
 -- Underhållspersonal
 -- Sök fram alla bilar i behov av kontroll
+SELECT vehicle.registrationNumber FROM green_rental.vehicle
+LEFT JOIN booking_details ON vehicle.registrationNumber = booking_details.registrationNumber
+LEFT JOIN booking ON booking_details.bookingNumber = booking.bookingNumber
+LEFT JOIN control ON vehicle.registrationNumber=control.registrationNumber
+WHERE control.controlLarge is FALSE and control.datum < "2024-04-12" and booking.endDate = "2024-04-12";
+
 -- Sök fram alla bilar i behov av kontroll inom 3/6/12 månader
+SELECT vehicle.registrationNumber
+FROM green_rental.vehicle
+LEFT JOIN control ON control.registrationNumber=vehicle.registrationNumber
+WHERE control.controlLarge is TRUE and control.datum>=DATE_SUB(CURDATE(), INTERVAL 3 MONTH);
+
 -- Sök fram alla bilar som har en skada
+SELECT vehicle.RegistrationNumber, damage.damageID, damage.descriptionDamage
+FROM green_rental.vehicle
+LEFT JOIN control ON control.registrationNumber=vehicle.registrationNumber
+LEFT JOIN damage ON damage.controlID=control.controlID
+WHERE damageID is not NULL AND damage.repairedDate is NULL;
+
 -- Lägg till en skada
+INSERT INTO green_rental.damage (controlID,fixedDamage,repairedDate,descriptionDamage)
+SELECT c.controlID,
+FALSE as fixedDamage,
+NULL as repairedDate,
+"Stenskott" as descriptionDamage
+FROM green_rental.control as c
+WHERE c.registrationNumber="ABC123";
+
 -- Lägg till en kontroll
+INSERT INTO green_rental.control (registrationNumber,staffID,controlDate,controlLarge,fuelLevel)
+SELECT v.registrationNumber,
+2 as staffID,
+b.endDate as controlDate,
+FALSE as controlLarge,
+30 as fuelLevel
+FROM green_rental.vehicle as v
+LEFT JOIN booking_details bd ON v.registrationNumber=bd.registrationNumber
+LEFT JOIN booking b ON bd.bookingNumber=b.bookingNumber
+WHERE b.endDate="2024-04-12";
```

Comment:
This is likely not needed.

Reply:
Replaced with manually added values instead